pgactivity/types.py

```diff
@@ -685,28 +716,79 @@ def default(
         cls,
         *,
         memory: Optional[MemoryInfo] = None,
-        swap: Optional[MemoryInfo] = None,
+        swap: Optional[SwapInfo] = None,
         load: Optional[LoadAverage] = None,
     ) -> "SystemInfo":
         """Zero-value builder.
 
         >>> SystemInfo.default()  # doctest: +NORMALIZE_WHITESPACE
-        SystemInfo(memory=MemoryInfo(percent=0.0, used=0, total=0),
-                   swap=MemoryInfo(percent=0.0, used=0, total=0),
+        SystemInfo(memory=MemoryInfo(used=0, buff_cached=0, free=0, total=0),
+                   swap=SwapInfo(used=0, free=0, total=0),
                    load=LoadAverage(avg1=0.0, avg5=0.0, avg15=0.0),
                    io_read=IOCounter(count=0, bytes=0, chars=0),
-                   io_write=IOCounter(count=0, bytes=0, chars=0),
-                   max_iops=0)
+                   io_write=IOCounter(count=0, bytes=0, chars=0), max_iops=0)
         """
         return cls(
             memory or MemoryInfo.default(),
-            swap or MemoryInfo.default(),
+            swap or SwapInfo.default(),
             load or LoadAverage.default(),
             IOCounter.default(),
             IOCounter.default(),
+            0,
         )
 
 
+@attr.s(auto_attribs=True, slots=True)
+class ServerInformations:
+    active_connections: int
+    idle: int
+    idle_in_transaction: int
+    idle_in_transaction_aborted: int
+    total: int
+    max_connections: int
+    av_workers: int
+    av_max: int
+    logical_workers: int
+    parallel_workers: int
+    walsenders: int
+    walreceivers: int
+    max_logical_workers: int
+    max_parallel_workers: int
+    max_worker_processes: int
+    max_wal_senders: int
+    max_replication_slots: int
+    replication_slots: int
+    xact_count: int
+    total_size: int
+    size_evolution: float
+    cache_hit_ratio: Pct = attr.ib(converter=Pct)
+    timestamp: int  # an epoch,  used for the calculation of the tps & size_evolution
+    uptime: timedelta
+    tps: int
+    insert: int
+    update: int
+    delete: int
+    tuples_returned: int
+    insert_per_second: int
+    update_per_second: int
+    delete_per_second: int
+    tuples_returned_per_second: int
+    temp_files: int
+    temp_bytes: int
+    max_dbname_length: int
+    waiting: int
+    worker_processes: int = attr.ib(init=False)
```

Comment:
I guess this is not very safe since the class is not Frozen ? or is it ?

Reply:
Yes, this should be a property. Better if the class can be made frozen too.